### PR TITLE
feat: add rocksdb statistics to metrics endpoint

### PR DESCRIPTION
**Pull request by hanabi1224**
**Summary of changes**
Changes introduced in this pull request:
- Add `rocksdb` statistics to metrics endpoint at ~~http://localhost:6116/metrics~~ http://localhost:6116/stats/db

Sample output of rockdb section on mainnet
```
# RocksDB statistics:
rocksdb.block.cache.miss COUNT : 1189335
rocksdb.block.cache.hit COUNT : 276005
rocksdb.block.cache.add COUNT : 428544
rocksdb.block.cache.add.failures COUNT : 0
rocksdb.block.cache.index.miss COUNT : 0
rocksdb.block.cache.index.hit COUNT : 0
rocksdb.block.cache.index.add COUNT : 0
rocksdb.block.cache.index.bytes.insert COUNT : 0
rocksdb.block.cache.index.bytes.evict COUNT : 0
rocksdb.block.cache.filter.miss COUNT : 0
rocksdb.block.cache.filter.hit COUNT : 0
rocksdb.block.cache.filter.add COUNT : 0
rocksdb.block.cache.filter.bytes.insert COUNT : 0
rocksdb.block.cache.filter.bytes.evict COUNT : 0
rocksdb.block.cache.data.miss COUNT : 1189335
rocksdb.block.cache.data.hit COUNT : 276005
rocksdb.block.cache.data.add COUNT : 428544
rocksdb.block.cache.data.bytes.insert COUNT : 2068761801
rocksdb.block.cache.bytes.read COUNT : 1362540705
rocksdb.block.cache.bytes.write COUNT : 2068761801
rocksdb.bloom.filter.useful COUNT : 1002966
rocksdb.bloom.filter.full.positive COUNT : 503202
rocksdb.bloom.filter.full.true.positive COUNT : 493317
rocksdb.bloom.filter.micros COUNT : 0
rocksdb.persistent.cache.hit COUNT : 0
rocksdb.persistent.cache.miss COUNT : 0
rocksdb.sim.block.cache.hit COUNT : 0
rocksdb.sim.block.cache.miss COUNT : 0
rocksdb.memtable.hit COUNT : 56158
rocksdb.memtable.miss COUNT : 693755
rocksdb.l0.hit COUNT : 6
rocksdb.l1.hit COUNT : 142970
rocksdb.l2andup.hit COUNT : 550491
rocksdb.compaction.key.drop.new COUNT : 288273
rocksdb.compaction.key.drop.obsolete COUNT : 0
rocksdb.compaction.key.drop.range_del COUNT : 0
rocksdb.compaction.key.drop.user COUNT : 0
rocksdb.compaction.range_del.drop.obsolete COUNT : 0
rocksdb.compaction.optimized.del.drop.obsolete COUNT : 0
rocksdb.compaction.cancelled COUNT : 0
rocksdb.number.keys.written COUNT : 200315
rocksdb.number.keys.read COUNT : 749913
rocksdb.number.keys.updated COUNT : 0
rocksdb.bytes.written COUNT : 208095544
rocksdb.bytes.read COUNT : 649401172
rocksdb.number.db.seek COUNT : 0
rocksdb.number.db.next COUNT : 0
rocksdb.number.db.prev COUNT : 0
rocksdb.number.db.seek.found COUNT : 0
rocksdb.number.db.next.found COUNT : 0
rocksdb.number.db.prev.found COUNT : 0
rocksdb.db.iter.bytes.read COUNT : 0
rocksdb.no.file.closes COUNT : 0
rocksdb.no.file.opens COUNT : 24820
rocksdb.no.file.errors COUNT : 0
rocksdb.l0.slowdown.micros COUNT : 0
rocksdb.memtable.compaction.micros COUNT : 0
rocksdb.l0.num.files.stall.micros COUNT : 0
rocksdb.stall.micros COUNT : 0
rocksdb.db.mutex.wait.micros COUNT : 0
rocksdb.rate.limit.delay.millis COUNT : 0
rocksdb.num.iterators COUNT : 0
rocksdb.number.multiget.get COUNT : 0
rocksdb.number.multiget.keys.read COUNT : 0
rocksdb.number.multiget.bytes.read COUNT : 0
rocksdb.number.deletes.filtered COUNT : 0
rocksdb.number.merge.failures COUNT : 0
rocksdb.bloom.filter.prefix.checked COUNT : 0
rocksdb.bloom.filter.prefix.useful COUNT : 0
rocksdb.number.reseeks.iteration COUNT : 0
rocksdb.getupdatessince.calls COUNT : 0
rocksdb.block.cachecompressed.miss COUNT : 0
rocksdb.block.cachecompressed.hit COUNT : 0
rocksdb.block.cachecompressed.add COUNT : 0
rocksdb.block.cachecompressed.add.failures COUNT : 0
rocksdb.wal.synced COUNT : 0
rocksdb.wal.bytes COUNT : 208095544
rocksdb.write.self COUNT : 9485
rocksdb.write.other COUNT : 0
rocksdb.write.timeout COUNT : 0
rocksdb.write.wal COUNT : 9485
rocksdb.compact.read.bytes COUNT : 3078540534
rocksdb.compact.write.bytes COUNT : 2930530440
rocksdb.flush.write.bytes COUNT : 0
rocksdb.compact.read.marked.bytes COUNT : 0
rocksdb.compact.read.periodic.bytes COUNT : 0
rocksdb.compact.read.ttl.bytes COUNT : 0
rocksdb.compact.write.marked.bytes COUNT : 0
rocksdb.compact.write.periodic.bytes COUNT : 0
rocksdb.compact.write.ttl.bytes COUNT : 0
rocksdb.number.direct.load.table.properties COUNT : 14
rocksdb.number.superversion_acquires COUNT : 74
rocksdb.number.superversion_releases COUNT : 7
rocksdb.number.superversion_cleanups COUNT : 7
rocksdb.number.block.compressed COUNT : 580964
rocksdb.number.block.decompressed COUNT : 835254
rocksdb.number.block.not_compressed COUNT : 147663
rocksdb.merge.operation.time.nanos COUNT : 0
rocksdb.filter.operation.time.nanos COUNT : 0
rocksdb.row.cache.hit COUNT : 0
rocksdb.row.cache.miss COUNT : 0
rocksdb.read.amp.estimate.useful.bytes COUNT : 0
rocksdb.read.amp.total.read.bytes COUNT : 0
rocksdb.number.rate_limiter.drains COUNT : 0
rocksdb.number.iter.skip COUNT : 0
rocksdb.blobdb.num.put COUNT : 0
rocksdb.blobdb.num.write COUNT : 0
rocksdb.blobdb.num.get COUNT : 0
rocksdb.blobdb.num.multiget COUNT : 0
rocksdb.blobdb.num.seek COUNT : 0
rocksdb.blobdb.num.next COUNT : 0
rocksdb.blobdb.num.prev COUNT : 0
rocksdb.blobdb.num.keys.written COUNT : 0
rocksdb.blobdb.num.keys.read COUNT : 0
rocksdb.blobdb.bytes.written COUNT : 0
rocksdb.blobdb.bytes.read COUNT : 0
rocksdb.blobdb.write.inlined COUNT : 0
rocksdb.blobdb.write.inlined.ttl COUNT : 0
rocksdb.blobdb.write.blob COUNT : 0
rocksdb.blobdb.write.blob.ttl COUNT : 0
rocksdb.blobdb.blob.file.bytes.written COUNT : 0
rocksdb.blobdb.blob.file.bytes.read COUNT : 0
rocksdb.blobdb.blob.file.synced COUNT : 0
rocksdb.blobdb.blob.index.expired.count COUNT : 0
rocksdb.blobdb.blob.index.expired.size COUNT : 0
rocksdb.blobdb.blob.index.evicted.count COUNT : 0
rocksdb.blobdb.blob.index.evicted.size COUNT : 0
rocksdb.blobdb.gc.num.files COUNT : 0
rocksdb.blobdb.gc.num.new.files COUNT : 0
rocksdb.blobdb.gc.failures COUNT : 0
rocksdb.blobdb.gc.num.keys.overwritten COUNT : 0
rocksdb.blobdb.gc.num.keys.expired COUNT : 0
rocksdb.blobdb.gc.num.keys.relocated COUNT : 0
rocksdb.blobdb.gc.bytes.overwritten COUNT : 0
rocksdb.blobdb.gc.bytes.expired COUNT : 0
rocksdb.blobdb.gc.bytes.relocated COUNT : 0
rocksdb.blobdb.fifo.num.files.evicted COUNT : 0
rocksdb.blobdb.fifo.num.keys.evicted COUNT : 0
rocksdb.blobdb.fifo.bytes.evicted COUNT : 0
rocksdb.txn.overhead.mutex.prepare COUNT : 0
rocksdb.txn.overhead.mutex.old.commit.map COUNT : 0
rocksdb.txn.overhead.duplicate.key COUNT : 0
rocksdb.txn.overhead.mutex.snapshot COUNT : 0
rocksdb.txn.get.tryagain COUNT : 0
rocksdb.number.multiget.keys.found COUNT : 0
rocksdb.num.iterator.created COUNT : 0
rocksdb.num.iterator.deleted COUNT : 0
rocksdb.block.cache.compression.dict.miss COUNT : 0
rocksdb.block.cache.compression.dict.hit COUNT : 0
rocksdb.block.cache.compression.dict.add COUNT : 0
rocksdb.block.cache.compression.dict.bytes.insert COUNT : 0
rocksdb.block.cache.compression.dict.bytes.evict COUNT : 0
rocksdb.block.cache.add.redundant COUNT : 17
rocksdb.block.cache.index.add.redundant COUNT : 0
rocksdb.block.cache.filter.add.redundant COUNT : 0
rocksdb.block.cache.data.add.redundant COUNT : 17
rocksdb.block.cache.compression.dict.add.redundant COUNT : 0
rocksdb.files.marked.trash COUNT : 0
rocksdb.files.deleted.immediately COUNT : 48
rocksdb.error.handler.bg.errro.count COUNT : 0
rocksdb.error.handler.bg.io.errro.count COUNT : 0
rocksdb.error.handler.bg.retryable.io.errro.count COUNT : 0
rocksdb.error.handler.autoresume.count COUNT : 0
rocksdb.error.handler.autoresume.retry.total.count COUNT : 0
rocksdb.error.handler.autoresume.success.count COUNT : 0
rocksdb.memtable.payload.bytes.at.flush COUNT : 0
rocksdb.memtable.garbage.bytes.at.flush COUNT : 0
rocksdb.secondary.cache.hits COUNT : 0
rocksdb.verify_checksum.read.bytes COUNT : 0
rocksdb.backup.read.bytes COUNT : 0
rocksdb.backup.write.bytes COUNT : 0
rocksdb.remote.compact.read.bytes COUNT : 0
rocksdb.remote.compact.write.bytes COUNT : 0
rocksdb.hot.file.read.bytes COUNT : 0
rocksdb.warm.file.read.bytes COUNT : 0
rocksdb.cold.file.read.bytes COUNT : 0
rocksdb.hot.file.read.count COUNT : 0
rocksdb.warm.file.read.count COUNT : 0
rocksdb.cold.file.read.count COUNT : 0
rocksdb.last.level.read.bytes COUNT : 0
rocksdb.last.level.read.count COUNT : 0
rocksdb.non.last.level.read.bytes COUNT : 10838416871
rocksdb.non.last.level.read.count COUNT : 1225855
rocksdb.block.checksum.compute.count COUNT : 1245945
rocksdb.multiget.coroutine.count COUNT : 0
rocksdb.db.get.micros P50 : 32.357917 P95 : 429.046129 P99 : 834.179487 P100 : 155906.000000 COUNT : 749913 SUM : 98196254
rocksdb.db.write.micros P50 : 14.378736 P95 : 1170.181661 P99 : 4211.136364 P100 : 73549.000000 COUNT : 9485 SUM : 2482604
rocksdb.compaction.times.micros P50 : 5400000.000000 P95 : 9428750.000000 P99 : 9562266.000000 P100 : 9562266.000000 COUNT : 9 SUM : 46901182
rocksdb.compaction.times.cpu_micros P50 : 4850000.000000 P95 : 8591367.000000 P99 : 8591367.000000 P100 : 8591367.000000 COUNT : 9 SUM : 39334772
rocksdb.subcompaction.setup.times.micros P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.table.sync.micros P50 : 116605.000000 P95 : 116605.000000 P99 : 116605.000000 P100 : 116605.000000 COUNT : 1 SUM : 116605
rocksdb.compaction.outfile.sync.micros P50 : 114285.714286 P95 : 203892.000000 P99 : 203892.000000 P100 : 203892.000000 COUNT : 49 SUM : 5399957
rocksdb.wal.file.sync.micros P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.manifest.file.sync.micros P50 : 4100.000000 P95 : 6201.000000 P99 : 6201.000000 P100 : 6201.000000 COUNT : 10 SUM : 42978
rocksdb.table.open.io.micros P50 : 348.510329 P95 : 732.219917 P99 : 4555.294118 P100 : 155761.000000 COUNT : 24820 SUM : 12756641
rocksdb.db.multiget.micros P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.read.block.compaction.micros P50 : 12.534917 P95 : 45.795242 P99 : 311.283702 P100 : 71272.000000 COUNT : 760791 SUM : 18854410
rocksdb.read.block.get.micros P50 : 171.789187 P95 : 372.174278 P99 : 678.558051 P100 : 155665.000000 COUNT : 460320 SUM : 74645432
rocksdb.write.raw.block.micros P50 : 2.681646 P95 : 5.092771 P99 : 9.151587 P100 : 231918.000000 COUNT : 728772 SUM : 9345487
rocksdb.l0.slowdown.count P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.memtable.compaction.count P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.num.files.stall.count P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.hard.rate.limit.delay.count P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.soft.rate.limit.delay.count P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.numfiles.in.singlecompaction P50 : 6.400000 P95 : 9.000000 P99 : 9.000000 P100 : 9.000000 COUNT : 9 SUM : 51
rocksdb.db.seek.micros P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.db.write.stall P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.sst.read.micros P50 : 3.902126 P95 : 268.890480 P99 : 458.394726 P100 : 119634.000000 COUNT : 1225856 SUM : 72388786
rocksdb.num.subcompactions.scheduled P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.bytes.per.read P50 : 368.652099 P95 : 2693.021374 P99 : 2876.890297 P100 : 35210.000000 COUNT : 749914 SUM : 649401496
rocksdb.bytes.per.write P50 : 468.280330 P95 : 275191.652470 P99 : 359215.502555 P100 : 1001453.000000 COUNT : 9485 SUM : 208095544
rocksdb.bytes.per.multiget P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.bytes.compressed P50 : 4815.879598 P95 : 6561.792556 P99 : 9108.234824 P100 : 789428.000000 COUNT : 580964 SUM : 2870057887
rocksdb.bytes.decompressed P50 : 4840.685423 P95 : 8330.921731 P99 : 421065.005163 P100 : 789428.000000 COUNT : 835255 SUM : 12976077985
rocksdb.compression.times.nanos P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.decompression.times.nanos P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.read.num.merge_operands P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.blobdb.key.size P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.blobdb.value.size P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.blobdb.write.micros P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.blobdb.get.micros P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.blobdb.multiget.micros P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.blobdb.seek.micros P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.blobdb.next.micros P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.blobdb.prev.micros P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.blobdb.blob.file.write.micros P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.blobdb.blob.file.read.micros P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.blobdb.blob.file.sync.micros P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.blobdb.gc.micros P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.blobdb.compression.micros P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.blobdb.decompression.micros P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.db.flush.micros P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.sst.batch.size P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.num.index.and.filter.blocks.read.per.level P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.num.data.blocks.read.per.level P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.num.sst.read.per.level P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.error.handler.autoresume.retry.count P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.async.read.bytes P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.poll.wait.micros P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.prefetched.bytes.discarded P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 51 SUM : 0
rocksdb.multiget.io.batch.size P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
rocksdb.num.level.read.per.multiget P50 : 0.000000 P95 : 0.000000 P99 : 0.000000 P100 : 0.000000 COUNT : 0 SUM : 0
```

**Reference issue to close (if applicable)**

Closes https://github.com/ChainSafe/forest/issues/2187

**Other information and links**

### CODE REVIEW

File: node/db/src/rocks.rs [hunk 1]
```diff
@@ -76,11 +77,17 @@ impl RocksDb {
     where
         P: AsRef<Path>,
     {
-        let db_opts = Self::to_options(config);
+        let mut db_opts = Self::to_options(config);
+        db_opts.enable_statistics();
```
Comment (elmattic) [inline node/db/src/rocks.rs:81]: We can already enable rocksdb statistics using the toml config file:

```
[rocks_db]
create_if_missing = true
parallelism = 12
write_buffer_size = 268435456
max_open_files = 1024
compaction_style = "level"
compression_type = "lz4"
enable_statistics = false
...
```

So I would remove the call to `enable_statistics()`.

Comment (hanabi1224) [reply node/db/src/rocks.rs:81]: Ah i c. Should we enable it by default?

**Review by LesnyRumcajs — COMMENTED** [review]
Does this format work with Grafana? https://github.com/ChainSafe/forest/tree/main/.maintain/monitoring

File: node/db/src/rocks.rs (hunk 1 above)
Comment (elmattic) [reply node/db/src/rocks.rs]: If there is not too much performance overhead maybe?

**Comment by elmattic** [comment]
I can make some wish list of related rocksdb metrics we should care about (if that helps), so:
- general ones
- specific to the settings we arfe exposing

**Comment by hanabi1224** [comment]
> Does this format work with Grafana? https://github.com/ChainSafe/forest/tree/main/.maintain/monitoring

I don't think so. How abt putting it under a different endpoint like metrics/db

**Comment by hanabi1224** [comment]
> I can make some wish list of related rocksdb metrics we should care about 

That's will be great!  Should it be covered in a doc? Or is there a way to exclude metrics we don't care in code?

File: node/db/src/rocks.rs (hunk 1 above)
Comment (hanabi1224) [reply node/db/src/rocks.rs]: I cannot find it in doc, but I don't observe notable performance / mem usage different locally when running mainnet

**Comment by LesnyRumcajs** [comment]
> > Does this format work with Grafana? https://github.com/ChainSafe/forest/tree/main/.maintain/monitoring
> 
> I don't think so. How abt putting it under a different endpoint like metrics/db

A completely different endpoint is satisfactory. I would not put it under `metrics`, though; it gives a misconception that it is Grafana compatible/generated via Prometheus. `stats/db` would be safer.

It would be even better to parse this data somehow to display it with Grafana, but I guess this is a separate issue.

**Comment by hanabi1224** [comment]
>  `stats/db` would be safer.

Sounds better, fixed

File: node/db/src/rocks.rs (hunk 1 above)
Comment (elmattic) [reply node/db/src/rocks.rs]: > The overhead of statistics is usually small but non-negligible. We usually observe an overhead of 5%-10%.

See here: https://github.com/facebook/rocksdb/wiki/Statistics#stats-level-and-performance-costs

Comment (hanabi1224) [reply node/db/src/rocks.rs]: Thanks for the info, let me turn it off by default

**Comment by elmattic** [comment]
A first list of stats that I can see:

```
// total block cache misses
rocksdb.block.cache.miss
// # of times cache miss when accessing index block from block cache
rocksdb.block.cache.index.miss
// # of times cache miss when accessing index block from block cache
rocksdb.block.cache.filter.miss
// # of times cache hit when accessing data block from block cache
rocksdb.block.cache.data.miss
// total block cache hit
rocksdb.block.cache.hit
// # of times cache hit when accessing index block from block cache
rocksdb.block.cache.index.hit
// # of times cache hit when accessing filter block from block cache
rocksdb.block.cache.filter.hit
// # of times cache hit when accessing data block from block cache
rocksdb.block.cache.data.hit
// # of bytes read from cache
rocksdb.block.cache.bytes.read
// # of bytes written into cache
rocksdb.block.cache.bytes.write

// # of times bloom filter has avoided file reads, i.e., negatives
rocksdb.bloom.filter.useful
// # of times bloom FullFilter has not avoided the reads
rocksdb.bloom.filter.full.positive
// # of times bloom FullFilter has not avoided the reads and data actually exist
rocksdb.bloom.filter.full.true.positive

// # of memtable hits
rocksdb.memtable.hit
// # of memtable misses
rocksdb.memtable.miss

// # of Get() queries served by L0
rocksdb.l0.hit
// # of Get() queries served by L1
rocksdb.l1.hit
// # of Get() queries served by L2 and up
rocksdb.l2andup.hit

// Number of keys written to the database via the Put and Write call's
rocksdb.number.keys.written
// Number of Keys read
rocksdb.number.keys.read

// The number of uncompressed bytes issued by DB::Put(), DB::Delete(), DB::Merge(), and DB::Write()
rocksdb.bytes.written
// The number of uncompressed bytes read from DB::Get(). It could be either from memtables, cache, or table files.
rocksdb.bytes.read

rocksdb.no.file.closes
rocksdb.no.file.opens

// Bytes read during compaction
rocksdb.compact.read.bytes
// Bytes written during compaction
rocksdb.compact.write.bytes
// Bytes written during flush
rocksdb.flush.write.bytes
```

**Review by lemmih — APPROVED** [review]

**Review by elmattic — APPROVED** [review]

**Comment by lemmih** [comment]
@hanabi1224 Let's push this over the finish line?

**Comment by hanabi1224** [comment]
@lemmih I will try to resolve the conflict introduced by rocksdb feature in an elegant way and request another review.

File: node/db/src/lib.rs [hunk 2]
```diff
@@ -125,3 +125,10 @@ impl<BS: Store> Store for &BS {
         (*self).bulk_delete(keys)
     }
 }
+
+/// Traits for collecting DB stats
+pub trait DBStatistics {
```
Comment (hanabi1224) [inline node/db/src/lib.rs:130]: New trait introduced to avoid adding `rocksdb` feature to `forest_utils`

**Pull request merged**